kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java

```diff
@@ -218,6 +261,36 @@ private List<FileStatus> getWinningCommitFiles(Engine engine) {
         }
     }
 
+    /**
+     * Get the last commit timestamp of the table. For ICT enabled tables, this is the ICT of the
+     * latest winning transaction commit file. For non-ICT enabled tables, this is the modification
+     * time of the latest winning transaction commit file.
+     *
+     * @param engine                {@link Engine} instance to use
+     * @param lastWinningVersion    last winning version of the table
+     * @param winningCommits        list of winning transaction commit files
+     * @param winningCommitInfoOpt  winning commit info
+     * @return last commit timestamp of the table
+     */
+    private long getLastCommitTimestamp(
+            Engine engine,
+            long lastWinningVersion,
+            List<FileStatus> winningCommits,
+            Optional<CommitInfo> winningCommitInfoOpt) {
+        FileStatus lastWinningTxn = winningCommits.get(winningCommits.size() - 1);
+        long winningCommitTimestamp = -1L;
+        if (snapshot.getVersion(engine) == -1 ||
+                !IN_COMMIT_TIMESTAMPS_ENABLED.fromMetadata(snapshot.getMetadata())) {
+            winningCommitTimestamp = lastWinningTxn.getModificationTime();
+        } else {
+            winningCommitTimestamp = CommitInfo.getRequiredInCommitTimestamp(
+                    winningCommitInfoOpt,
+                    String.valueOf(lastWinningVersion),
```

Comment:
why convert to string?

Reply:
Just for the error message print

Reply:
The `getRequiredInCommitTimestamp` has some logic to check if `winningCommitInfoOpt` is empty and if it contains ict and raises different errors accordingly. So I think it's better to leave this function handle with the error messages?